experimenter/tests/integration/nimbus/test_demo_app_integration.py

```diff
@@ -0,0 +1,234 @@
+import pytest
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from nimbus.pages.experimenter.home import HomePage
+from nimbus.pages.experimenter.summary import SummaryPage
+
+
+def navigate_to_demo_app_frontend(selenium):
+    demo_app_url = "http://demo-app-frontend:3000/"
+    selenium.get(demo_app_url)
+
+
+def fill_and_send_form_data(selenium, client_id, context):
+    client_id_input = WebDriverWait(selenium, 10).until(
+        EC.presence_of_element_located((By.XPATH, "//input[@placeholder='Client ID']"))
+    )
+    context_input = WebDriverWait(selenium, 10).until(
+        EC.presence_of_element_located((By.XPATH, "//input[@placeholder='Context']"))
+    )
+
+    client_id_input.send_keys(client_id)
+    context_input.send_keys(context)
+
+
+def click_send_my_details(selenium):
+    send_details_button = WebDriverWait(selenium, 10).until(
+        EC.element_to_be_clickable(
+            (By.XPATH, "//button[contains(text(), 'Send My Details')]")
+        )
+    )
+    assert send_details_button.is_displayed() and send_details_button.is_enabled()
+    send_details_button.click()
+
+
+@pytest.mark.demo_app
+def test_create_new_rollout_approve_remote_settings_demo_app(
+    selenium,
+    experiment_url,
+    create_experiment,
+    kinto_client,
+    base_url,
+    experiment_name,
+):
+    # Launch a rollout with 100% population
+    reference_branch_value = '{"enabled": true, "something": "You are enrolled"}'
+    create_experiment(
+        selenium, is_rollout=True, reference_branch_value=reference_branch_value
+    ).launch_and_approve()
+
+    kinto_client.approve()
+
+    SummaryPage(selenium, experiment_url).open().wait_for_live_status()
+
+    home = HomePage(selenium, base_url).open()
+    assert True in [experiment_name in item.text for item in home.tables[0].experiments]
+
+    # demo app frontend, default displays "Not Enrolled"
+    navigate_to_demo_app_frontend(selenium)
+    result_text_element = WebDriverWait(selenium, 10).until(
+        EC.presence_of_element_located(
+            (By.XPATH, "//h1[contains(text(), 'Not Enrolled')]")
+        )
+    )
```

Comment:
It would be nice to wrap this in one of the selenium Page object wrappers.  We could do that as a followup optimization to this.

Reply:
sure let me note it down and can circle back later, great idea